File: 2215-find-difference-in-two-arrays/solution.py
def findDifference(nums1: list[int], nums2: list[int]) -> list[list[int]]:
    n1 = {}
    n2 = {}
    for i in nums1:
        if i not in n1:
            n1[i] = 0
        n1[i] = n1[i] + 1
    for i in nums2:
        if i not in n2:
            n2[i] = 0
        n2[i] = n2[i] + 1
    
    # left is in n1 but not n2, right is in n2 but not in n1
    a = [[],[]]
    for i in n1:
        if i not in n2:
            a[0].append(i)
    for i in n2:
        if i not in n1:
            a[1].append(i)
    return a

File: 2215-find-difference-in-two-arrays/solution.go
package main

func findDifference(nums1 []int, nums2 []int) [][]int {
    n1 := make(map[int]int)
    n2 := make(map[int]int)

    for _, i := range nums1 {
        n1[i]++
    }
    for _, i := range nums2 {
        n2[i]++
    }

    a := make([][]int, 2)
    for k, _ := range n1 {
        if n2[k] == 0 {
            a[0] = append(a[0], k)
        }
    }

    for k, _ := range n2 {
        if n1[k] == 0 {
            a[1] = append(a[1], k)
        }
    }

    return a
}